e2e/e2e-windows-scenario.sh

```diff
@@ -0,0 +1,176 @@
+}
+EOF
+
+echo "Scenario is $SCENARIO_NAME"
+jq --arg clientCrt "$clientCertificate" --arg vmssName $DEPLOYMENT_VMSS_NAME 'del(.KubeletConfig."--pod-manifest-path") | del(.KubeletConfig."--pod-max-pids") | del(.KubeletConfig."--protect-kernel-defaults") | del(.KubeletConfig."--tls-cert-file") | del(.KubeletConfig."--tls-private-key-file") | .ContainerService.properties.certificateProfile += {"clientCertificate": $clientCrt} | .PrimaryScaleSetName=$vmssName' nodebootstrapping_config.json > nodebootstrapping_config_for_windows.json
+jq -s '.[0] * .[1]' nodebootstrapping_config_for_windows.json scenarios/$SCENARIO_NAME/property-$SCENARIO_NAME.json > scenarios/$SCENARIO_NAME/nbc-$SCENARIO_NAME.json
+
+go test -run TestE2EWindows
+
+MC_WIN_VMSS_NAME=$(az vmss list -g $MC_RESOURCE_GROUP_NAME --query "[?contains(name, '$WINDOWS_NODEPOOL')]" -ojson | jq -r '.[0].name')
+VMSS_RESOURCE_Id=$(az resource show --resource-group $MC_RESOURCE_GROUP_NAME --name $MC_WIN_VMSS_NAME --resource-type Microsoft.Compute/virtualMachineScaleSets --query id --output tsv)
+
+az group export --resource-group $MC_RESOURCE_GROUP_NAME --resource-ids $VMSS_RESOURCE_Id --include-parameter-default-value > test.json
+WINDOWS_VNET=$(jq -c '.parameters | with_entries( select(.key|contains("vnet")))' test.json)
+WINDOWS_LOADBALANCER=$(jq -c '.parameters | with_entries( select(.key|contains("loadBalancers")))' test.json)
+WINDOWS_IDENTITY=$(jq -c '.resources[0] | with_entries( select(.key|contains("identity")))' test.json)
+NETWORK_PROPERTIES=$(jq -c '.resources[0].properties.virtualMachineProfile.networkProfile.networkInterfaceConfigurations[0] | with_entries( select(.key|contains("properties")))' test.json)
+CUSTOM_DATA=$(cat scenarios/$SCENARIO_NAME/$SCENARIO_NAME-cloud-init.txt)
+CSE_CMD=$(cat scenarios/$SCENARIO_NAME/$SCENARIO_NAME-cseCmd)
+
+jq --argjson JsonForVnet "$WINDOWS_VNET" \
+    --argjson JsonForLB "$WINDOWS_LOADBALANCER" \
+    --argjson JsonForIdentity "$WINDOWS_IDENTITY" \
+    --argjson JsonForNetwork "$NETWORK_PROPERTIES" \
+    --arg ValueForAdminPassword "$WINDOWS_PASSWORD" \
+    --arg ValueForCustomData "$CUSTOM_DATA" \
+    --arg ValueForCSECmd "$CSE_CMD" \
+    --arg ValueForVMSS "$DEPLOYMENT_VMSS_NAME" \
+    '.parameters += $JsonForVnet | .parameters += $JsonForLB | .resources[0] += $JsonForIdentity | .resources[0].properties.virtualMachineProfile.networkProfile.networkInterfaceConfigurations[0] += $JsonForNetwork | .resources[0].properties.virtualMachineProfile.osProfile.adminPassword=$ValueForAdminPassword | .resources[0].properties.virtualMachineProfile.osProfile.customData=$ValueForCustomData | .resources[0].properties.virtualMachineProfile.extensionProfile.extensions[0].properties.settings.commandToExecute=$ValueForCSECmd | .parameters.virtualMachineScaleSets_akswin30_name.defaultValue=$ValueForVMSS' \
+    windows_vmss_template.json > deployment.json
+
+set +e
+az deployment group create --resource-group $MC_RESOURCE_GROUP_NAME \
```

Comment:
It seems like that it will fail in running several tests in parallel.
ERROR: {"code": "DeploymentActive", "message": "Unable to edit or replace deployment 'deployment': previous deployment from '12/15/2022 2:43:41 AM' is still active (expiration time is '12/22/2022 2:43:35 AM').

Reply:
use vmss name to clarify the deployment.json and resolve this  problem.